src/Lib/Element.php

```diff
@@ -70,7 +82,7 @@ public function getElementName()
      */
     final protected function addElement(Element $element, $isInternal = true)
     {
-        $elementName = $element->getElementName();
+        $elementName = (string) $element->getElementName();
```

Comment:
Maybe not when it comes to aggregations. Better safe than sorry. That bug took ages to find!